codex-ui/src/vue/components/alert/AlertContainer.vue

```diff
@@ -0,0 +1,59 @@
+<template>
+  <div ref="alertRef">
+    <div
+      :class="[
+        $style.alert__container,
+        $style['alert__container--' + props.position]
+      ]"
+    >
+      <AlertTransition>
+        <BaseAlert
+          v-for="alert in alertStore"
+          :key="alert.id"
+          v-bind="alert"
+        />
+      </AlertTransition>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import BaseAlert from './BaseAlert.vue';
+import AlertTransition from './AlertTransition.vue';
+import { useAlert } from './core/useAlert';
+import type { AlertOptions } from './core/types';
+import { ALERT_CONTAINER_STYLES } from './core/constant';
+
+const props = withDefaults(defineProps<AlertOptions>(), {
+  id: ALERT_CONTAINER_STYLES.id,
+  position: ALERT_CONTAINER_STYLES.position,
+  content: ALERT_CONTAINER_STYLES.message,
+  icon: ALERT_CONTAINER_STYLES.icon,
+  type: ALERT_CONTAINER_STYLES.type,
+  timeout: ALERT_CONTAINER_STYLES.timeout,
+});
+
+const {
+  alertRef,
+  alertStore,
```

Comment:
there seems to be over complicated. See popover for example: https://github.com/codex-team/notes.web/blob/main/codex-ui/src/vue/components/popover/Popover.vue

Just return a list of current alerts:

```ts
const { alerts } = useAlerts()
```

and then render them through the `v-for`